sway/desktop/xdg_shell.c

```diff
@@ -152,10 +149,6 @@ static void destroy(struct sway_view *view) {
 	if (xdg_shell_view == NULL) {
 		return;
 	}
-	wl_list_remove(&xdg_shell_view->destroy.link);
```

Comment:
Hmm, why moving this in `handle_destroy`? This seems more confusing.

Reply:
Functions named `destroy` should free. This is a convention we use everywhere. Functions that don't free should have a different name.